functions/index.js

```diff
@@ -143,3 +143,49 @@ app.post('/', isAuthenticated ,function(req,response)
 });
 
 exports.signUp = functions.https.onRequest(app);
+
+
+
+//<--------Random Fact Generation---------->
+//Generates random fact about techspardha.
+exports.randomFact = functions.https.onRequest((request , response) => {
+  const numberOfLines = 8;
+  const randomIndex = Math.floor(Math.random() * numberOfLines);
+  database.ref('/facts/' + randomIndex).on('value',function(snapshot){
+    console.log(snapshot.val());
+    response.status(401).json({
+      message : snapshot.val()
+    });
+  });
+});
+
+
+//<------Returning the array of all the videos------>
+//Returns the array of videos containing title and url of a video.
+exports.getVideo = functions.https.onRequest((request , response) => {
+  let items = [];
+  database.ref('/videos').on('value', function(snapshot) {
+    snapshot.forEach(function(childSnapshot) {
+      items.push(childSnapshot.key,{
+        title : childSnapshot.val().title,
+        url : childSnapshot.val().url
+      });
+    });
+  });
+  response.status(401).json(items);
```

Comment:
I think directly returning snapshot.val() should work. no need to iterate and make separate object

Reply:
Yup, there is a function snapshotToArray(snapshot) for the same purpose.  we hadn't tried it yet, will do that.